From 82608848e2950a679c49a4a92cd7f6d50bf24141 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 16 Apr 2025 13:52:21 +0300
Subject: [PATCH] [SW-218933] g3: use access mode enum from specs

The `mem_access_mode` enum was defined within individual Zephyr apps. This
enum has been moved to specs dir to avoid definitions across multiple apps
in an earlier patch. This patch replaces the older enum with enum from
specs dir in fwloader agent

Change-Id: Id5c69d68c208a1d15dcad6cc1316d29073524a7b
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/common/include/security.h b/common/include/security.h
index bafcc4f..d82c9cf 100644
--- a/common/include/security.h
+++ b/common/include/security.h
@@ -15,12 +15,6 @@
 #define HL_SEC_CHECK_ENV "security_check"
 #define KEY_PROP_EXP_SZ		8
 
-enum fwloader_mem_access_mode {
-	MEM_ACCESS_RO,
-	MEM_ACCESS_WR,
-	MEM_ACCESS_RW,
-};
-
 /* This structure is suitable for any min/max based RRs */
 struct hl_sec_rr_cfg {
 	int rr;
@@ -46,7 +40,7 @@
 int hl_sec_bis_e_compliant_enable(void);
 bool hl_sec_is_bis_compliant(void);
 int hl_sec_erom_recovery_enable(void);
-void hl_sec_dyn_img_open(uint64_t off, uint32_t size, enum fwloader_mem_access_mode mode);
+void hl_sec_dyn_img_open(uint64_t off, uint32_t size, enum mem_access_mode mode);
 void hl_sec_dyn_img_close(void);
 void hl_sec_dynamic_init(void);
 int get_pub_key_node(const void *pub_key_fit, const char *key_name);
diff --git a/common/src/gaudi2/security.c b/common/src/gaudi2/security.c
index a1e8cc2..5e1ff86 100644
--- a/common/src/gaudi2/security.c
+++ b/common/src/gaudi2/security.c
@@ -321,7 +321,7 @@
 }
 
 #define SIZE_4K	0x1000
-void hl_sec_dyn_img_open(uint64_t off, uint32_t size, __unused enum fwloader_mem_access_mode mode)
+void hl_sec_dyn_img_open(uint64_t off, uint32_t size, __unused enum mem_access_mode mode)
 {
 	struct hl_sec_rr_cfg *cfg;
 	uint64_t img_max, img_full_addr;
diff --git a/common/src/gaudi3/security.c b/common/src/gaudi3/security.c
index 0ef105d..22f96ae 100644
--- a/common/src/gaudi3/security.c
+++ b/common/src/gaudi3/security.c
@@ -100,7 +100,7 @@
 }
 
 #define IMAGE_OFFSET_MASK	0xFFFFFF
-void hl_sec_dyn_img_open(uint64_t off, uint32_t size, enum fwloader_mem_access_mode mode)
+void hl_sec_dyn_img_open(uint64_t off, uint32_t size, enum mem_access_mode mode)
 {
 	uint64_t img_max, image_base_full;
 
@@ -130,7 +130,7 @@
 	 * RR5 is used to enforce the write protection on the SRAM region dynamically
 	 * opened to the host, as it is exposed with read-only (RO) permissions.
 	 */
-	if (mode == MEM_ACCESS_RO)
+	if (mode == ACCESS_RO)
 		hl_sec_config_dyn_rr(HL_SEC_HBW_RR_5, image_base_full, img_max, HL_SEC_WR_PROT);
 
 	/*
diff --git a/fw-loader/src/fw_upd.c b/fw-loader/src/fw_upd.c
index 0f68f55..29dce17 100644
--- a/fw-loader/src/fw_upd.c
+++ b/fw-loader/src/fw_upd.c
@@ -336,7 +336,7 @@
 		}
 
 		if (tmp_offset)
-			hl_sec_dyn_img_open((uintptr_t)tmp_offset, fwstate.res_size, MEM_ACCESS_RO);
+			hl_sec_dyn_img_open((uintptr_t)tmp_offset, fwstate.res_size, ACCESS_RO);
 	default:
 		/* on success the original offset is set,  */
 		fwupd_send_status_simple(sts, tmp_offset);
@@ -655,7 +655,7 @@
 		goto prog_timeout;
 
 	if (fwstate.res_size) {
-		hl_sec_dyn_img_open((uintptr_t)fwstate.buf, fwstate.res_size, MEM_ACCESS_RW);
+		hl_sec_dyn_img_open((uintptr_t)fwstate.buf, fwstate.res_size, ACCESS_RW);
 		fwupd_send_status(HL_UPD_STS_PROG_OK, fwstate.buf);
 	} else {
 		fwupd_send_status(HL_UPD_STS_PROG_OK, 0);
@@ -694,7 +694,7 @@
 
 	/* make sure the data is in RAM */
 	sys_cache_flush((void *)fwstate.buf, sz);
-	hl_sec_dyn_img_open((uintptr_t)fwstate.buf, sz, MEM_ACCESS_RW);
+	hl_sec_dyn_img_open((uintptr_t)fwstate.buf, sz, ACCESS_RW);
 
 	if (fwupd_wfc(WFC_TIMEOUT_MS))
 		goto map_timeout;
@@ -835,7 +835,7 @@
 		goto resp_end;
 	}
 
-	hl_sec_dyn_img_open((uintptr_t)fwstate.buf, fwstate.res_size, MEM_ACCESS_RO);
+	hl_sec_dyn_img_open((uintptr_t)fwstate.buf, fwstate.res_size, ACCESS_RO);
 	fwupd_send_status_simple(HL_UPD_STS_OK, fwstate.buf);
 
 	return CMD_RET_SUCCESS;
diff --git a/fw-loader/src/security.c b/fw-loader/src/security.c
index 0ce3790..6ffe262 100644
--- a/fw-loader/src/security.c
+++ b/fw-loader/src/security.c
@@ -179,7 +179,7 @@
 
 EMPTY_WEAK_FUNC(void hl_sec_dyn_img_close(void),)
 EMPTY_WEAK_FUNC(void hl_sec_dyn_img_open(uint64_t off, uint32_t size,
-		enum fwloader_mem_access_mode mode),)
+		enum mem_access_mode mode),)
 EMPTY_WEAK_FUNC(void hl_sec_dynamic_init(void),)
 EMPTY_WEAK_FUNC(int fit_image_signature_validation(const char *img_name, const void *data, size_t image_size), 0)
 
